libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -183,7 +183,9 @@ void AP_Arming::update(void)
     const uint32_t now_ms = AP_HAL::millis();
     // perform pre-arm checks & display failures every 30 seconds
     bool display_fail = false;
-    if (now_ms - last_prearm_display_ms > PREARM_DISPLAY_PERIOD*1000) {
+    if ((report_immediately && (now_ms - last_prearm_display_ms > 750)) ||
```

Comment:
Can you increase this to 3-5s so that we don't get rapid oscillation when a GPS is on the edge, say

Reply:
good idea. Done. Took the average of your suggestion, so 4 secs :)